docs/DEVSETUP.md

```diff
@@ -1,70 +1,181 @@
+
+## Setting Up
+
+It is advisable to work on this project using an IDE rather than through the github repository workspace; as this makes it easier to make changes to this project using git commands.
+
+Follow these next steps to setup your local repository in your IDE.
+
+## Clone this repository <a name="clone"></a>
+This simply means replicating the files in this project in your IDE workspace.
+
+### Step 1
+
+Navigate to your forked repository, on the left side of your screen click on the drop down with "code" written on it. This will pop up a small dialog box; make no changes to this box but simply copy the url.
+
+![](./Clone.png)
+
+
+### Step 2
+
+Create a folder on your computer and open this folder in your IDE. Open the terminal in your IDE and cd into this folder
+
+Run the command
+
+```
+git init
+```
+
+To clone the remote repository to your desktop repository, run
+
+```
+git clone "paste-the-link-you-copied-from-the-forked-repository"
+```
+
+for example,
+```
+git clone https://github.com/Chizobaonorh/codebooker.git
+```
+
+## Setup up working environment<a name="env"></a>
+
+After cloning this repository, create your feature branch where you commit changes you make before merging into the main branch.
+
+### Step 1
+
+Create your feature branch using
 
-The page will reload when you make changes.\
-You may also see any lint errors in the console.
+```
+git checkout -b "feature-branch"
+```
 
-### `npm test`
+### Step 2
 
-Launches the test runner in the interactive watch mode.\
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+ After this, install dependencies by running
 
-### `npm run build`
+ ```
+ npm i
 
-Builds the app for production to the `build` folder.\
-It correctly bundles React in production mode and optimizes the build for the best performance.
+ ```
 
-The build is minified and the filenames include the hashes.\
-Your app is ready to be deployed!
+Next run,
 
+```
+npm start
+```
+
+Runs the app in the development mode. 
+Open [http://localhost:3000](http://localhost:3000) to view it in your browser. The page will reload when you make changes. You may also see any lint errors in the console.
+
+Next,
+
+```
+npm test
+```
+
+Launches the test runner in the interactive watch mode. See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+
+Then, 
+
+```
+npm run build
+```
+
+Builds the app for production to the `build` folder. It correctly bundles React in production mode and optimizes the build for the best performance.
+
+The build is minified and the filenames include the hashes. Your app is ready to be deployed!
 See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
```

Comment:
facebook.github.io links are all now redirected to: https://opensource.fb.com/ so anything referencing any facebook.github.io link can be removed becuase those links in this document were the original create-react-app docs.

Reply:
"npm run buid" while still in development ?

Reply:
yeah this would not be good either.  This is not a step we are ready to do until we are ready for deployment.  This part can be removed.

I think it would be ok to merge this, but a 2nd round of documentation cleanup on this might need to be done just to be sure we are thorough and clear on steps, etc.

@Chizobaonorh @LOGESH-B @BlackBond06

Reply:
I agree. When the suggested changes are made, then it should be merged.

Reply:
Yes and we have the same sort of issues with the documentation on ReactSocialNetwork.

Reply:
> "npm run buid" while still in development ?

I followed the steps for setting up that was in the devsetup page before i edited it. But i will get rid of it.

Reply:
Oh okay. That command is ran when the project is ready for deployment.